compiling error

make install
Making install in src/ucm
make[1]: Entering directory '/home/jinchen/TEST/ucx-1.11.2/src/ucm'
Making install in .
make[2]: Entering directory '/home/jinchen/TEST/ucx-1.11.2/src/ucm'
  CC       event/libucm_la-event.lo
  CC       malloc/libucm_la-malloc_hook.lo
  CC       mmap/libucm_la-install.lo
  CC       util/libucm_la-replace.lo
  CC       util/libucm_la-log.lo
  CC       util/libucm_la-reloc.lo
  CC       util/libucm_la-sys.lo
  CC       bistro/libucm_la-bistro.lo
  CC       bistro/libucm_la-bistro_x86_64.lo
  CC       bistro/libucm_la-bistro_aarch64.lo
  CC       bistro/libucm_la-bistro_ppc64.lo
"bistro/bistro_ppc64.c", line 157: error #1144: arithmetic on pointer to void or function type
      func += 8;
           ^

"bistro/bistro_ppc64.c", line 158: error #1144: arithmetic on pointer to void or function type
      hook += 8;
           ^

2 errors detected in the compilation of "bistro/bistro_ppc64.c".

@dmitrygx Thanks. it's fixed. But now I'm getting the following errors. Do you have any idea?

make[2]: Entering directory '/home/jinchen/TEST/ucx-1.11.2/src/tools/info'
  CC       ucx_info-build_info.o
  CC       ucx_info-proto_info.o
  CC       ucx_info-sys_info.o
  CC       ucx_info-tl_info.o
  CC       ucx_info-type_info.o
  CC       ucx_info-ucx_info.o
  CCLD     ucx_info
ucx_info-proto_info.o: In function `print_ucp_info':
/home/jinchen/TEST/ucx-1.11.2/src/tools/info/proto_info.c:66: undefined reference to `__builtin_ppc_get_timebase'
/home/jinchen/TEST/ucx-1.11.2/src/tools/info/proto_info.c:66: undefined reference to `__builtin_ppc_get_timebase'
/home/jinchen/TEST/ucx-1.11.2/src/tools/info/proto_info.c:83: undefined reference to `__builtin_ppc_get_timebase'
/home/jinchen/TEST/ucx-1.11.2/src/tools/info/proto_info.c:83: undefined reference to `__builtin_ppc_get_timebase'
ucx_info-sys_info.o: In function `print_sys_info':
/home/jinchen/TEST/ucx-1.11.2/src/tools/info/sys_info.c:72: undefined reference to `__builtin_ppc_get_timebase'
ucx_info-sys_info.o:/home/jinchen/TEST/ucx-1.11.2/src/tools/info/sys_info.c:75: more undefined references to `__builtin_ppc_get_timebase' follow
make[2]: *** [Makefile:563: ucx_info] Error 2
make[2]: Leaving directory '/home/jinchen/TEST/ucx-1.11.2/src/tools/info'
make[1]: *** [Makefile:783: install] Error 2
make[1]: Leaving directory '/home/jinchen/TEST/ucx-1.11.2/src/tools/info'
make: *** [Makefile:735: install-recursive] Error 1

I did followed #7619, but the error is still there. So I followed your instruction one more time to be sure. And I'm still getting them

make  install-am
make[2]: Entering directory '/TEST/ucx/src/tools/info'
  CC       ucx_info-build_info.o
  CC       ucx_info-proto_info.o
  CC       ucx_info-sys_info.o
  CC       ucx_info-tl_info.o
  CC       ucx_info-type_info.o
  CC       ucx_info-ucx_info.o
  CCLD     ucx_info
ucx_info-proto_info.o: In function `print_ucp_info':
/home/jinchen/TEST/ucx/src/tools/info/proto_info.c:66: undefined reference to `__builtin_ppc_get_timebase'
/home/jinchen/TEST/ucx/src/tools/info/proto_info.c:66: undefined reference to `__builtin_ppc_get_timebase'
/home/jinchen/TEST/ucx/src/tools/info/proto_info.c:83: undefined reference to `__builtin_ppc_get_timebase'
/home/jinchen/TEST/ucx/src/tools/info/proto_info.c:83: undefined reference to `__builtin_ppc_get_timebase'
ucx_info-sys_info.o: In function `print_sys_info':
/home/jinchen/TEST/ucx/src/tools/info/sys_info.c:73: undefined reference to `__builtin_ppc_get_timebase'
ucx_info-sys_info.o:/home/jinchen/TEST/ucx/src/tools/info/sys_info.c:76: more undefined references to `__builtin_ppc_get_timebase' follow

--- a/config/m4/sysdep.m4
+++ b/config/m4/sysdep.m4
@@ -96,10 +96,33 @@ AC_CHECK_DECLS([ethtool_cmd_speed, SPEED_UNKNOWN], [], [],
 
 
 #
-# PowerPC query for TB frequency
+# PowerPC "sys/platform/ppc.h" header
 #
-AC_CHECK_DECLS([__ppc_get_timebase_freq], [], [], [#include <sys/platform/ppc.h>])
 AC_CHECK_HEADERS([sys/platform/ppc.h])
+
+
+#
+# PowerPC query for getting TB frequency
+#
+AC_CHECK_DECL([__ppc_get_timebase_freq],
+              [AC_CHECK_FUNCS([__ppc_get_timebase_freq])], [],
+              [#include <sys/platform/ppc.h>])
+
+
+#
+# PowerPC query for getting TB.
+# Note: AC_CHECK_FUNCS doesn't work for checking __ppc_get_timebase()
+#
+AC_LINK_IFELSE([AC_LANG_SOURCE([[
+                #include <sys/platform/ppc.h>
+                int main(int argc, char** argv) {
+                    __ppc_get_timebase();
+                    return 0;
+                } ]])],
+                [AC_MSG_RESULT([no])
+                 AC_DEFINE([HAVE___PPC_GET_TIMEBASE], [1],
+                           [__ppc_get_timebase is defined in ppc.h])],
+                [AC_MSG_RESULT([no])])
 
 
 #

--- a/src/ucm/bistro/bistro_ppc64.c
+++ b/src/ucm/bistro/bistro_ppc64.c
@@ -154,8 +154,8 @@ ucs_status_t ucm_bistro_patch_toc(void *func_ptr, void *hook,
     }
 
 #if defined(_CALL_ELF) && (_CALL_ELF == 2)
-    func += 8;
-    hook += 8;
+    func = UCS_PTR_BYTE_OFFSET(func, 8);
+    hook = UCS_PTR_BYTE_OFFSET(hook, 8);
 #endif
 
     ucm_bistro_fill_patch(&patch, R11, (uintptr_t)hook);

--- a/src/ucs/arch/ppc64/cpu.h
+++ b/src/ucs/arch/ppc64/cpu.h
@@ -41,12 +41,12 @@ BEGIN_C_DECLS
 
 static inline uint64_t ucs_arch_read_hres_clock()
 {
-#ifndef HAVE_SYS_PLATFORM_PPC_H
+#ifdef HAVE___PPC_GET_TIMEBASE
+    return __ppc_get_timebase();
+#else
     uint64_t tb;
     asm volatile ("mfspr %0, 268" : "=r" (tb));
     return tb;
-#else
-    return __ppc_get_timebase();
 #endif
 }
 

--- a/src/ucs/arch/ppc64/timebase.c
+++ b/src/ucs/arch/ppc64/timebase.c
@@ -19,7 +19,7 @@
 
 double ucs_arch_get_clocks_per_sec()
 {
-#if HAVE_DECL___PPC_GET_TIMEBASE_FREQ
+#ifdef HAVE___PPC_GET_TIMEBASE_FREQ
     return __ppc_get_timebase_freq();
 #else
     return ucs_get_cpuinfo_clock_freq("timebase", 1.0);

